python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -0,0 +1,978 @@
+            fiberId=originalFiberId,
+            wavelength=wavelength,
+            flux=flux,
+            metadata={},
+            fitFlag=fitFlagArray,
+            fitFlagNames=self.fitFlagNames,
+            fitParams=fitParams,
+        )
+
+    def getRadialVelocities(self, pfsConfig, pfsMerged, bbPdfs):
+        """Estimate the radial velocity for each fiber in ``pfsMerged``.
+
+        Parameters
+        ----------
+        pfsConfig : `pfs.datamodel.pfsConfig.PfsConfig`
+            Configuration of the PFS top-end.
+        pfsMerged : `pfs.datamodel.pfsFiberArraySet.PfsFiberArraySet`
+            Typically an instance of PsfMerged.
+            It must have been whitened.
+        bbPdfs : `List[Optional[numpy.array]]`
+            `bbPdfs[i]`, if not None, is the probability distribution
+            of `pfsConfig.fiberId[i]` being of each model type,
+            determined by broad-band photometries.
+
+        Returns
+        -------
+        radialVelocities : `List[Optional[lsst.pipe.base.Struct]]`
+            Radial velocity for each fiber.
+            Each element, if not None, has `velocity` and `error`
+            as its member. See ``EstimateRadialVelocityTask``.
+        """
+        # Find the best model from broad bands.
+        # This model is used as the reference for cross-correlation calculation
+        bestModels = self.findRoughlyBestModel(bbPdfs)
+
+        radialVelocities = []
+        for spectrum, model in zip(fibers(pfsConfig, pfsMerged), bestModels):
+            if model.spectrum is None:
+                radialVelocities.append(None)
+                continue
+            modelSpectrum = convolveLsf(model.spectrum)
+            modelSpectrum = self.whitenSpectrum(modelSpectrum, mode="model")
+            radialVelocities.append(self.estimateRadialVelocity.run(spectrum, modelSpectrum))
+
+        return radialVelocities
+
+    def whitenSpectrum(self, spectra, *, mode):
+        """Whiten one or more spectra.
+
+        Parameters
+        ----------
+        spectra : `PfsSimpleSpectrum` or `PfsFiberArraySet`
+            spectra to whiten.
+        mode : `str`
+            "observed" or "model".
+            Whether ``spectra`` is from observation or from simulation.
+
+        Returns
+        -------
+        spectra : `PfsSimpleSpectrum` or `PfsFiberArraySet`
+            The same instance as the argument.
+        """
+        if mode == "observed":
+            fitContinuum = self.fitObsContinuum
+        if mode == "model":
+            fitContinuum = self.fitModelContinuum
+
+        # If `spectra` is actually a single spectrum,
+        # we put it into PfsFiberArraySet
+        if len(spectra.flux.shape) == 1:
+            original_spectrum = spectra
+            if not hasattr(spectra, "covar"):
+                spectra = promoteSimpleSpectrumToFiberArray(spectra, snr=400)
+            spectra = promoteFiberArrayToFiberArraySet(spectra, fiberId=1)
```

Comment:
The result object of these promotions is used only temporarily. `fiberId` can be any number. I chose "1" because I was afraid that `fiberId=0` might possibly have some special meaning.